src/app/use-cases/validate-document.ts

```diff
@@ -18,25 +18,51 @@ import {
   DocumentList,
   IsDocumentUploaded,
 } from "../../signature-request/document";
-import { dispatch } from "./status-signature-request";
 
 import { GetDocumentPayload } from "./get-document";
+import { dispatchOnDocument, DocumentAction } from "./status-document";
 
+const documentNotFoundError = new EntityNotFoundError("Document not found");
 export const addUrlToDocument =
   (documentId: Document["id"], url: string) => (request: SignatureRequest) =>
     pipe(
       request.documents,
-      findIndex((document) => document.id === documentId),
+      A.findIndex((document) => document.id === documentId),
       O.chain((index) =>
         pipe(
           request.documents,
-          modifyAt(index, (document) => ({
+          A.modifyAt(index, (document) => ({
             ...document,
             url,
           }))
         )
       ),
-      E.fromOption(() => new EntityNotFoundError("Document not found")),
+      E.fromOption(() => documentNotFoundError),
+      E.chainW(validate(DocumentList, "Unable to validate document list"))
+    );
+
+export const changeDocumentStatus =
+  (documentId: Document["id"], action: DocumentAction) =>
+  (request: SignatureRequest) =>
+    pipe(
+      request.documents,
+      A.findIndex((document) => document.id === documentId),
+      E.fromOption(() => documentNotFoundError),
+      E.chainW((index) =>
+        pipe(
+          request.documents,
+          A.lookup(index),
+          E.fromOption(() => documentNotFoundError),
+          E.chain(dispatchOnDocument(action)),
+          E.chain((document) =>
+            pipe(
+              request.documents,
+              A.modifyAt(index, () => document),
+              E.fromOption(() => documentNotFoundError)
```

Comment:
you don't nee this line

Reply:
`A.modifyAt(index, () => document)` change an element in the array and return an option. So the check on the option must be done anyway